frontend/src/api/Socket.ts

```diff
@@ -12,11 +12,18 @@ export type User = {
 // Variable to hold the current connected state.
 let connected: boolean = false;
 
-// Create constants for the connection, subscription, and send message URLs.
-export const SOCKET_ENDPOINT:string = '/api/v1/socket/join-room-endpoint';
-export const SUBSCRIBE_URL:string = '/api/v1/socket/subscribe-user';
-const ADD_USER_URL:string = '/api/v1/socket/add-user';
-const DELETE_USER_URL:string = '/api/v1/socket/delete-user';
+// Dynamic route endpoints that depend on the room id
+const basePath = '/api/v1/socket';
+let socketRoomId: string;
+export const routes = (roomId: string) => {
+  socketRoomId = roomId;
```

Comment:
Why is this included here? The `socketRoomId` is reset every time the `routes(...)` is called, but `socketRoomId` is the parameter for this method often?

Reply:
Yeah so socketRoomId is meant to save the latest roomId passed in since sometimes the routes are used within `Socket.ts` and it needs to know what roomId to use. It's an imperfect solution so I'd be open to changing it if you had any suggestions

Reply:
If you think creating a React hook would be cleaner, you could do that, but otherwise I think it's alright.